code/modules/mining/lavaland/loot/hierophant_loot.dm

```diff
@@ -49,6 +50,14 @@
 
 /obj/item/hierophant_club/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
 	..()
+	if(!is_mining_level(user.z)) //Will only spawn a few sparks if not on mining z level
+
+		if(world.time > timer)
```

Comment:
Instead of checking the timer twice in this proc it's best to move line 62 to the front. The timer check only that is. And remove this one.
so something like this:
```
if(!(timer check))
 return

if(!mining)
 your stuff
 return

var/T = turf
if(!T)
 return
other stuff
```

Reply:
It makes sense, but I'm not confident enough with this syntax to do that modification and assure the rest will work the same way...

Reply:
Can always ask somebody in coding chat to help you. Or check it before you commit it.
Enough people there willing to help!

Reply:
I think I solved it!